maestro-client/src/main/java/maestro/drivers/AndroidDriver.kt

```diff
@@ -400,9 +400,27 @@ class AndroidDriver(
     }
 
     override fun openLink(link: String) {
-        dadb.shell("am start -a android.intent.action.VIEW -d \"$link\"")
+        if (link.startsWith(HTTPS_SCHEME) || link.startsWith(HTTP_SCHEME)) {
+            val installedPackages = installedPackages()
+            when {
+                installedPackages.contains("com.android.chrome") -> {
+                    dadb.shell("am start -a android.intent.action.VIEW -d \"$link\" com.android.chrome")
+                }
+                installedPackages.contains("org.mozilla.firefox") -> {
```

Comment:
@dmitry-zaitsev should we be checking for more browser apps?

Reply:
I think that is enough for now, let's see what users say once we roll this out